types/rdf-js/index.d.ts

```diff
@@ -615,14 +615,14 @@ export interface Dataset<Q extends BaseQuad = Quad> extends DatasetCore<Q> {
     /**
      * Returns a new `Dataset` that is a concatenation of this dataset and the quads given as an argument.
      */
-    union(quads: Dataset<Q>): this;
+    union(quads: Dataset<BaseQuad>): this;
 }
 
-export interface DatasetFactory<Q extends BaseQuad = Quad> extends DatasetCoreFactory<Q> {
+export interface DatasetFactory<Q extends BaseQuad = Quad, D extends Dataset<Q> = Dataset<Q>> extends DatasetCoreFactory<Q, D> {
     /**
      * Returns a new dataset and imports all quads, if given.
      */
-    dataset(quads?: Dataset<Q>|Q[]): Dataset<Q>;
+    dataset(quads?: DatasetCore<BaseQuad>|BaseQuad[]): D;
```

Comment:
While I agree that it should be `DatasetCore` rather than `Dataset`, the spec does say `Dataset`. (Though given there's quite a few mistakes in it...)

Reply:
Fair enough, it does say `Dataset`

Reply:
And I have not change `Dataset#addAll` which has similar signature so best keep consistent.

Reply:
Ideally this would just be an `Iterable`, but outside the scope here.

Reply:
Had that thought but that would deviate greatly from the spec